### PR TITLE
DD_APPSEC_SCA_ENABLED env var test

### PR DESCRIPTION
**Pull request by gnufede**
## Motivation

This PR enables the check for `DD_APPSEC_SCA_ENABLED` env variable. This variable and it's value is sent through telemetry if set, otherwise it's not sent.

See RFC: https://docs.google.com/document/d/1xTLC3UEGNooZS0YOYp3swMlAhtvVn1aa639TGxHHYvg/edit

Jira: APPSEC-52424

## Changes

Two more parametric tests for the app-started event, one with the variable set, other with it unset.

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
    * [ ] To R&P team: locally build and push the image to hub.docker.com 
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
    * [ ] Once merged, add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

File: tests/parametric/test_telemetry.py [hunk 1]
```diff
@@ -254,3 +254,86 @@ def test_telemetry_event_not_propagated(self, library_env, test_agent, test_libr
...
+            assert (
+                "configuration" in body["payload"]
+            ), "The configuration should be included in the telemetry event, got {}".format(body)
+
+            configuration = body["payload"]["configuration"]
+
+            configuration_by_name = {item["name"]: item for item in configuration}
+
+            DD_APPSEC_SCA_ENABLED = "DD_APPSEC_SCA_ENABLED"
+            if context.library in ("dotnet", "jvm", "nodejs"):
```
Comment (robertomonteromiguel) [inline tests/parametric/test_telemetry.py:298]: jvm?

File: tests/parametric/test_telemetry.py [hunk 2]
```diff
@@ -254,3 +254,86 @@ def test_telemetry_event_not_propagated(self, library_env, test_agent, test_libr
...
```
Comment (robertomonteromiguel) [inline tests/parametric/test_telemetry.py:334]: jvm?

File: tests/parametric/test_telemetry.py [hunk 3]
```diff
@@ -254,3 +254,86 @@ def test_telemetry_event_not_propagated(self, library_env, test_agent, test_libr
...
+
+    @pytest.mark.parametrize("library_env", [{**DEFAULT_ENVVARS}])
+    def test_telemetry_sca_enabled_not_propagated(self, library_env, test_agent, test_library):
+        with test_library.start_span("first_span"):
+            pass
+
+        test_agent.wait_for_telemetry_event("app-started")
+        requests = test_agent.raw_telemetry(clear=True)
+        assert len(requests) > 0, "There should be at least one telemetry event (app-started)"
+        for req in requests:
```
Comment (robertomonteromiguel) [inline tests/parametric/test_telemetry.py:321]: Perhaps some of the duplicate logic could be eliminated by copying it to a common method.

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]

File: tests/parametric/test_telemetry.py [hunk 4]
```diff
@@ -254,3 +254,77 @@ def test_telemetry_event_not_propagated(self, library_env, test_agent, test_libr
...
+    )
+    def test_telemetry_sca_enabled_propagated(self, library_env, outcome_value, test_agent, test_library):
+        configuration_by_name = self.get_app_started_configuration_by_name(library_env, test_agent, test_library)
+
+        DD_APPSEC_SCA_ENABLED = self.get_dd_appsec_sca_enabled_str(context.library)
+
+        cfg_appsec_enabled = configuration_by_name.get(DD_APPSEC_SCA_ENABLED)
+        assert cfg_appsec_enabled is not None, "Missing telemetry config item for '{}'".format(DD_APPSEC_SCA_ENABLED)
+
+        if context.library in ("golang", "java", "dotnet"):
```
Comment (CarlesDD) [inline tests/parametric/test_telemetry.py:319]:
```suggestion
        if context.library in ("golang", "java", "dotnet", "nodejs"):
```

**Review by CarlesDD — CHANGES_REQUESTED** [review]

**Review by robertomonteromiguel — APPROVED** [review]

File: tests/parametric/test_telemetry.py [hunk 5]
```diff
@@ -254,3 +254,77 @@ def test_telemetry_event_not_propagated(self, library_env, test_agent, test_libr
...
+            configuration_by_name = {item["name"]: item for item in configuration}
+            return configuration_by_name
+
+        return None
+
+    @staticmethod
+    def get_dd_appsec_sca_enabled_str(library):
+        DD_APPSEC_SCA_ENABLED = "DD_APPSEC_SCA_ENABLED"
+        if library in ("dotnet", "java", "nodejs"):
+            DD_APPSEC_SCA_ENABLED = "appsec.sca.enabled"
```
Comment (smola) [inline tests/parametric/test_telemetry.py:295]: This one seems to be missing at https://github.com/DataDog/dd-go/blob/prod/trace/apps/tracer-telemetry-intake/telemetry-payload/static/config_norm_rules.json#L30

Comment (gnufede) [reply tests/parametric/test_telemetry.py:295]: Here it is: https://github.com/DataDog/dd-go/pull/130968

**Review by ygree — APPROVED** [review]
LGTM

File: tests/parametric/test_telemetry.py [hunk 6]
```diff
@@ -254,3 +254,77 @@ def test_telemetry_event_not_propagated(self, library_env, test_agent, test_libr
...
+            return configuration_by_name
+
+        return None
+
+    @staticmethod
+    def get_dd_appsec_sca_enabled_str(library):
+        DD_APPSEC_SCA_ENABLED = "DD_APPSEC_SCA_ENABLED"
+        if library in ("java", "nodejs"):
+            DD_APPSEC_SCA_ENABLED = "appsec.sca.enabled"
+        elif library == "php":
```
Comment (vpellan) [inline tests/parametric/test_telemetry.py:296]:
```suggestion
        elif library in ("php", "ruby"):
```

File: tests/parametric/test_telemetry.py [hunk 7]
```diff
@@ -254,3 +254,77 @@ def test_telemetry_event_not_propagated(self, library_env, test_agent, test_libr
...
+    )
+    def test_telemetry_sca_enabled_propagated(self, library_env, outcome_value, test_agent, test_library):
+        configuration_by_name = self.get_app_started_configuration_by_name(library_env, test_agent, test_library)
+
+        DD_APPSEC_SCA_ENABLED = self.get_dd_appsec_sca_enabled_str(context.library)
+
+        cfg_appsec_enabled = configuration_by_name.get(DD_APPSEC_SCA_ENABLED)
+        assert cfg_appsec_enabled is not None, "Missing telemetry config item for '{}'".format(DD_APPSEC_SCA_ENABLED)
+
+        if context.library in ("golang", "java", "dotnet", "nodejs"):
```
Comment (vpellan) [inline tests/parametric/test_telemetry.py:319]:
```suggestion
        if context.library in ("golang", "java", "dotnet", "nodejs", "ruby"):
```

**Review by vpellan — CHANGES_REQUESTED** [review]

**Review by vpellan — APPROVED** [review]

**Review by NachoEchevarria — APPROVED** [review]
LGTM. Thanks!

**Review by ValentinZakharov — APPROVED** [review]

**Review by gillarramendi — APPROVED** [review]

**Pull request merged**